src/app/data/data.service.ts

```diff
@@ -22,25 +23,97 @@ export class DataService {
   get dataLevels() { return DataLevels; }
   get dataAttributes() { return DataAttributes; }
   get bubbleAttributes() { return BubbleAttributes; }
-  activeYear = 2015;
+  activeYear;
   activeFeatures: MapFeature[] = [];
   activeDataLevel: MapLayerGroup = DataLevels[0];
   activeDataHighlight: MapDataAttribute = DataAttributes[0];
   activeBubbleHighlight: MapDataAttribute = BubbleAttributes[0];
   mapView;
-  mapConfig = {
-    style: './assets/style.json',
-    center: [-98.5556199, 39.8097343],
-    zoom: 3,
-    minZoom: 3,
-    maxZoom: 14
-  };
+  mapConfig;
   private mercator = new SphericalMercator({ size: 256 });
   private tileBase = 'https://s3.us-east-2.amazonaws.com/eviction-lab-tilesets/fixtures/';
   private tilePrefix = 'evictions-';
   private tilesetYears = ['90', '00', '10'];
-  private maxZoom = 10;
-  constructor(private http: Http) { }
+  private queryZoom = 10;
+
+  constructor(private http: Http) {}
+
+  /**
+   * Sets the choropleth layer based on the provided `DataAttributes` ID
+   * @param id string corresponding to the `MapDataAttribute` in `DataAttributes`
+   */
+  setChoroplethHighlight(id: string) {
+    const dataAttr = this.dataAttributes.find((attr) => attr.id === id);
+    if (dataAttr) {
+      this.activeDataHighlight = dataAttr;
+    }
+  }
+
+  /**
+   * Sets the bubble layer based on the provided `BubbleAttributes` ID
+   * @param id string corresponding to the `MapDataAttribute` in `BubbleAttributes`
+   */
+  setBubbleHighlight(id: string) {
+    const bubbleAttr = this.bubbleAttributes.find((attr) => attr.id === id);
+    if (bubbleAttr) {
+      this.activeBubbleHighlight = bubbleAttr;
+    }
+  }
+
+  /**
+   * Sets the layer geography based on the provided `DataLevels` ID
+   * @param id string of the MapLayerGroup in `DataLevels`
+   */
+  setGeographyLevel(id: string) {
+    const geoLevel = this.dataLevels.find((level) => level.id === id);
+    if (geoLevel) {
+      this.activeDataLevel = geoLevel;
+    }
+  }
+
+  /** */
+  setLocations(locations) {
+    locations.forEach(l => {
+      this.getTileData(l.layer, l.lonLat, true)
+        .subscribe((data) => { this.addLocation(data); });
+    });
+  }
+
+  /**
+   * Sets the bounding box for the map to focus to
+   * @param mapBounds an array with four coordinates representing south, west, north, east
+   */
+  setMapBounds(mapBounds) {
+    this.mapView = mapBounds;
+  }
+
+  /**
+   * Returns the URL parameters for the current view
+   */
+  getUrlParameters() {
+    const paramMap = [ 'locations', 'year', 'geography', 'type', 'choropleth', 'bounds' ];
+    return this.getRouteArray().reduce((a, b, i) => {
+      return a + ';' + paramMap[i] + '=' + b;
+    }, '');
+  }
+
+  /**
+   * Gets an array of values that represent the current route
+   */
+  getRouteArray() {
+    const locations = this.activeFeatures.map((f, i, arr) => {
+      const xy = this.getFeatureXY(f);
+      return f.properties['layerId'] + ',' + xy.x + ',' + xy.y;
```

Comment:
Would it be easier to just have the center point latitude and longitude of a feature here? My worry is that X and Y are zoom-dependent, so if there are any other uses to this than just pulling the data it could be safer to use the lat lon coordinates

Reply:
Yeah, that's what I was thinking initially.  You had suggested using X/Y values because it's shorter and less processing, which made sense to me.  Are there concerns that we wouldn't get the proper feature using X/Y?

Reply:
Ahh whoops, my bad! If this was in reference to the exports, that library you sent me before cleared up any issues with using lat lon. And if it's just for getting tile data then no, but not sure if this would play into anything else